modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java

```diff
@@ -0,0 +1,139 @@
+package com.enonic.xp.core.impl.project;
+
+import java.util.Map;
+import java.util.concurrent.Executors;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.enonic.xp.content.ContentConstants;
+import com.enonic.xp.content.ContentService;
+import com.enonic.xp.context.Context;
+import com.enonic.xp.context.ContextBuilder;
+import com.enonic.xp.core.internal.concurrent.SimpleExecutor;
+import com.enonic.xp.event.Event;
+import com.enonic.xp.event.EventListener;
+import com.enonic.xp.media.MediaInfoService;
+import com.enonic.xp.project.ParentProjectSynchronizer;
+import com.enonic.xp.project.Project;
+import com.enonic.xp.project.ProjectName;
+import com.enonic.xp.project.ProjectService;
+import com.enonic.xp.security.PrincipalKey;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.User;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+
+@Component(immediate = true)
+public class ProjectEventListener
+    implements EventListener
+{
+    private final static Logger LOG = LoggerFactory.getLogger( ProjectEventListener.class );
```

Comment:
Codacy found an issue: ['static' modifier out of order with the JLS suggestions.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6117594)